lib/user_comm_screener.rb

```diff
@@ -193,6 +237,20 @@ def load_preference_map
     UserCommPrefs.new(acting_user, resolved_user_communication_preferences)
   end
 
+  def actor_preferences
+    @actor_preferences ||= begin
+      user_ids_by_preference_type = actor_communication_preferences.reduce(Hash.new([])) do |hash, pref|
```

Comment:
```suggestion
      user_ids_by_preference_type = actor_communication_preferences.reduce(Hash.new { [] }) do |hash, pref|
```

Oh for reference data types like arrays we need to pass a block that returns the desired default value to `Hash.new` otherwise all keys would share the same array:

```
irb(main):001:0> a = Hash.new([])
=> {}
irb(main):002:0> a[:a] << 1
=> [1]
irb(main):003:0> a[:b]
=> [1]
```

vs

```
irb(main):004:0> a = Hash.new { [] }
=> {}
irb(main):005:0> a[:a] << 1
=> [1]
irb(main):006:0> a[:b]
=> []
```

Reply:
Thanks @OsamaSayegh , commited that suggestion